src/cpu/reorder/simple_sparse_reorder.hpp

```diff
@@ -55,10 +55,9 @@ namespace cpu {
 // - dense_tag -> encoding
 // - encoding -> dense_tag
 #define SIMPLE_SPARSE_REORDER_TEMPL_DECL \
-    impl::data_type_t type_i, typename fmt_i_t, fmt_i_t fmt_i, \
-            impl::data_type_t type_o, typename fmt_o_t, fmt_o_t fmt_o
-#define SIMPLE_SPARSE_REORDER_TEMPL_CALL \
-    type_i, fmt_i_t, fmt_i, type_o, fmt_o_t, fmt_o
+    impl::data_type_t type_i, impl::format_tag_t fmt_i, \
```

Comment:
Actually, we need to get the original version back. This reorder is also supposed to be used to reorder data between dense format and sparse format that could be described with `format_tag_t` and `sparse_encoding_t` respectively. Because of that we need the template to have the data type parameters for input/output formats. We also have to use SFINAE to disable/enable certain implementations based on the data type, which is why constexpr `is_format_tag` function is used for that.